src/cli_utils.cc

```diff
@@ -234,84 +183,89 @@ Status unified_sites_of_yaml(const YAML::Node& yaml,
     return Status::OK();
 }
 
+Status merge_all(const vector<discovered_alleles> &valleles,
+                 discovered_alleles &ans) {
+    Status s;
+    ans.clear();
+    for (auto it = valleles.begin(); it != valleles.end(); ++it) {
+        S(merge_discovered_alleles(*it, ans));
+    }
+    return Status::OK();
+}
+
 
 // Load first file
 Status merge_discovered_allele_files(std::shared_ptr<spdlog::logger> logger,
                                      const vector<string> &filenames,
                                      vector<pair<string,size_t>> &contigs,
-                                     vector<range> &ranges,
-                                     vector<discovered_alleles> &valleles) {
+                                     discovered_alleles &dsals) {
     Status s;
     if (filenames.size() == 0)
         return Status::Invalid("no discovered allele files provided");
     contigs.clear();
-    ranges.clear();
-    valleles.clear();
+    dsals.clear();
 
     // Load the first file
     YAML::Node node;
     const string& first_file = filenames[0];
     S(LoadYAMLFile(first_file, node));
-    S(contigs_alleles_ranges_of_yaml(node, contigs, ranges, valleles));
-    log_num_alleles(logger, valleles, first_file);
+    S(contigs_alleles_of_yaml(node, contigs, dsals));
+    logger->info() << "loaded " << dsals.size() << " alleles from " << first_file;
 
     if (filenames.size() == 1)
         return Status::OK();
 
-    // We need to merge additional files. Make a map from range to
-    // discovered-allele structures. Each discovered alleles file
-    // may have a different set of containing ranges. We want to merge alleles
-    // only when they are in the same containing range.
-    map<range, discovered_alleles> range2dsal;
-    for (int j=0; j < ranges.size(); ++j) {
-        range2dsal[ranges[j]] = std::move(valleles[j]);
-    }
-
-    // extra hygiene, the loop below fills these arrays
-    ranges.clear();
-    valleles.clear();
-
     // Load the rest of the files, and merge
-    for (int i=1; i < filenames.size(); ++i) {
+    for (auto it = filenames.begin() + 1; it != filenames.end(); ++it) {
         vector<pair<string,size_t>> contigs2;
-        vector<range> ranges2;
-        vector<discovered_alleles> valleles2;
+        discovered_alleles dsals2;
         YAML::Node node;
-        const string &crnt_file = filenames[i];
+        const string &crnt_file = *it;
 
         S(LoadYAMLFile(crnt_file, node));
-        S(contigs_alleles_ranges_of_yaml(node, contigs2, ranges2, valleles2));
-        log_num_alleles(logger, valleles2, crnt_file);
+        S(contigs_alleles_of_yaml(node, contigs2, dsals2));
+        logger->info() << "loaded " << dsals2.size() << " alleles from " << crnt_file;
 
-        // sanity: verify that the contigs are the same
+        // verify that the contigs are the same
```

Comment:
I think you can just test `contigs != contigs2` http://www.cplusplus.com/reference/vector/vector/operators/